integrationtest/src/it/scala/io/gearpump/integrationtest/checklist/ConnectorKafkaSpec.scala

```diff
@@ -55,19 +62,71 @@ class ConnectorKafkaSpec extends TestSpecBase {
         "-brokerList", brokerList,
         "-sourceTopic", sourceTopic,
         "-sinkTopic", sinkTopic).mkString(" ")
-      val success = restClient.submitApp(jar, args)
-      Thread.sleep(5000)
-      success shouldEqual true
+      restClient.submitApp(jar, args)
       val actual = restClient.queryApp(appId)
 
       actual.appId shouldEqual appId
       actual.status shouldEqual "active"
       actual.appName shouldEqual "KafkaReadWrite"
 
-      val messageNum = 10000
-      kafkaCluster.produceDataToKafka(zookeeperConnect, brokerList, sourceTopic, messageNum)
-      kafkaCluster.getLatestOffset(brokerList, sinkTopic) shouldBe messageNum
+      Util.retryUntil(kafkaCluster.getLatestOffset(brokerList, sinkTopic) == messageNum)
+    }
+  }
+
+  "Gearpump with Kafka" should {
+    "support at least once" in {
+      val zookeeperConnect = s"${kafkaCluster.getHostname}:${kafkaCluster.getZookeeperPort}"
+      val brokerList = s"${kafkaCluster.advertised_host}:${kafkaCluster.advertised_port}"
+      val sourcePartitionNum = 2
+      val sourceTopic = "topic3"
+      val sinkTopic = "topic4"
+      val appsCount = restClient.listApps().size
+      val appId = appsCount + 1
+
+      //Generate number sequence to the topic, start from number 1
+      kafkaCluster.createTopic(zookeeperConnect, sourceTopic, sourcePartitionNum)
+      val producer = new NumericalDataProducer(sourceTopic, brokerList)
+      producer.start()
+
+      // exercise
+      val jar = cluster.queryBuiltInExampleJars("kafka-").head
+      val args = Array("io.gearpump.streaming.examples.kafka.KafkaReadWrite",
+        "-zookeeperConnect", zookeeperConnect,
+        "-brokerList", brokerList,
+        "-sourceTopic", sourceTopic,
+        "-sinkTopic", sinkTopic,
+        "-source", sourcePartitionNum).mkString(" ")
+      restClient.submitApp(jar, args)
+
+      val actual = restClient.queryApp(appId)
+      actual.appId shouldEqual appId
+
+      Util.retryUntil(restClient.queryStreamingAppDetail(appId).clock > 0)
+      val executorToKill = restClient.getExecutorBrief(appId).map(_.executorId).max
+      restClient.killExecutor(appId, executorToKill) shouldBe true
+      producer.stop()
```

Comment:
I found the test spec that extends from ```TestSpecBase``` could not have ```before``` and ```after``` because the parent already have, and can not override it.